### PR TITLE
Workflow to create a pre-release on GitHub

### PR DESCRIPTION
**Pull request by SoaAlex**
Creates a pre-release on the associated GitHub repository.

Runs just after the workflow to prepare for next development version, therefore only one workflow manual trigger is necessary.

Uploads artifacts to release from the previous workflow.

Uses the same tag for release as the one created in previous workflow.

It's the same workflow for tools and components (only the last 2 actions are specific for tools to upload artifacts).

See this example: https://github.com/riseclipse/riseclipse-validator-scl2003/releases/tag/riseclipse-validator-scl2003-1.2.0
See this PR for SCL Validator to apply for this tool: https://github.com/riseclipse/riseclipse-validator-scl2003/pull/45

Note: workflow trigger will be changed to `workflow_call` once PR approved, it was just for testing.

### CODE REVIEW

File: .github/workflows/Shared-Pre-Release-GH.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    runs-on: ubuntu-latest
+    name: Deploy pre-release
+
+    steps:
+    - name: Checkout ${{ github.event.repository.name }}
+      uses: actions/checkout@v3
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
```
Comment (marcadetd) [inline .github/workflows/Shared-Pre-Release-GH.yml:47]: This action is unmaintained

File: .github/workflows/Shared-Pre-Release-GH.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+name: Create pre-release on GitHub
+
+# This CD workflow creates a pre-release on GitHub
+# It aslo uploads artifacts if it's a tool
+
+on:
+  workflow_call:
+    inputs:
+      isTool:
```
Comment (marcadetd) [inline .github/workflows/Shared-Pre-Release-GH.yml:28]: I am not sure to have a use-case for pre-releasing a component (yagni)

File: .github/workflows/Shared-Pre-Release-GH.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.event.repository.name }}-${{ inputs.release_version }}
+        release_name: ${{ github.event.repository.name }} v${{ inputs.release_version }}
+        body: This RiseClipse release contains the latest changes available. You can learn more by reading the documentation.
```
Comment (marcadetd) [inline .github/workflows/Shared-Pre-Release-GH.yml:53]: This pre-release contains the latest build of this RiseClipse tooL

**Comment by marcadetd** [comment]
What happens if I run again this workflow?

File: .github/workflows/Shared-Pre-Release-GH.yml (hunk 1 above)
Comment (SoaAlex) [reply .github/workflows/Shared-Pre-Release-GH.yml:47]: Will take a look and test this action

File: .github/workflows/Shared-Pre-Release-GH.yml (hunk 2 above)
Comment (SoaAlex) [reply .github/workflows/Shared-Pre-Release-GH.yml:28]: If there is no need, I can delete this option. Workflow remains nearly unchanged

**Comment by SoaAlex** [comment]
> What happens if I run again this workflow?

Currently, it will crash but not because of the second workflow but because of the first one to prepare for next development version.
The reason is that you have to delete the PR, delete the tag and the do_branch in order to be able to re run the 1st workflow (`prepare-next-dev`) again.
The second workflow (`create-pre-release`) would be able to run again if you do these previous steps.

File: .github/workflows/Shared-Pre-Release-GH.yml (hunk 2 above)
Comment (marcadetd) [reply .github/workflows/Shared-Pre-Release-GH.yml:28]: yes, delete

**Comment by marcadetd** [comment]
The release will be done also on github.io and docker, so it can be done also on GH. The pre-releases can be deleted by hand.

**Comment by SoaAlex** [comment]
> If I rerun it, it it because a correction has been done, therefore the branch and tag will have been deleted
Then, if you also delete the pre-release, It will run flawlessly (it might also work if pre-release wasn't deleted but I didn't test it).

**Comment by SoaAlex** [comment]
> Let's keep it simple: you do either a pre-release or a release. Both will call next-dev to get the artifacts. Therefore, it is during the pre-release job, and after next-dev, that the suffix is added.

OK, that's looks also the best solution for me.

I will try to add the suffix during the pre-release job. 
One way of doing it right now is to execute a shell command on the workflow that will rename all assets in the artifacts folder by adding `rc`.
This doesn't seem difficult to do.

**Comment by SoaAlex** [comment]
Was able to achieve all requirements

See this pre-release: https://github.com/riseclipse/riseclipse-validator-scl2003/releases

File: .github/workflows/Shared-Pre-Release-GH.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      uses: actions/checkout@v3
+
+    - name: Download artifacts
+      uses: actions/download-artifact@v3
+      with:
+        path: ${{ github.workspace }}/artifacts
+
+    - name: Add rc suffix to artifacts
+      run: |
+        sudo apt-get install -y rename
```
Comment (marcadetd) [inline .github/workflows/Shared-Pre-Release-GH.yml:48]: should be done in a "Setup Environment" step

**Comment by SoaAlex** [comment]
Made suggested changes:
https://github.com/riseclipse/riseclipse-validator-scl2003/releases

**Review by marcadetd — APPROVED** [review]

**Pull request merged**